src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,66 @@
+package duke;
+
+import java.lang.reflect.Constructor;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import duke.command.Command;
+import duke.exception.InvalidInputException;
+import duke.exception.InvalidInputException.InputExceptionType;
+
+public class Parser {
```

Comment:
You might want to include a Javadoc comment explaining what this class does

Reply:
Thanks for the advice. This has been fixed in the latest commit.